src/openforms/appointments/tasks.py

```diff
@@ -21,37 +18,20 @@
     ignore_result=False,
     once={"graceful": True},  # do not spam error monitoring
 )
-def maybe_register_appointment(submission_id: int) -> None | str:
+def register_appointment(submission_id: int) -> None | str:
```

Comment:
I'd stick with the original task name though, the `maybe_` prefix indicates that the task itself may not do anything at all. This is done so that the logic to act or not act is contained in the task itself (in this task `NoAppointmentForm` can be raised and then nothing is done).

The location where the task is scheduled then reads more natural:

* maybe we need to register an appointment, then
* maybe run pre-registration, then
* (definitely) generate the summary PDF, then
* maybe run registration, then
* maybe send a confirmation email

etc.

From the scheduling point of view, you don't know what will run or not, and while you can check it in there, that greatly breaks the loose coupling because you end up with a "god function" that understands the whole system. We don't do that, instead we set up the chain of tasks to execute and use naming conventions to indicate what will or might happen.